src/bmconfigparser.py

```diff
@@ -142,6 +144,15 @@ def safeGetInt(self, section, field, default=0):
                 ValueError, AttributeError):
             return default
 
+    def safeGetFloat(self, section, field, default=0.0):
+        """Return value as float, default on exceptions,
+        0.0 if default missing"""
+        try:
+            return self.getfloat(section, field)
+        except (configparser.NoSectionError, configparser.NoOptionError,
+                ValueError, AttributeError):
+            return default
+
```

Comment:
You need a test for this in `test_config`

Reply:
I agree

Reply:
Added.